### PR TITLE
fix(replay): validate UUIDs for get_replay_counts and OrganizationReplayCountEndpoint

### PR DESCRIPTION
**Pull request by aliu39**
Fixes [SENTRY-2HS0](https://sentry.sentry.io/issues/4930519003/)

Linked in @ryan953 to fix the issue of invalid frontend queries. Related to https://github.com/getsentry/sentry/issues/65887.

For better visibility, invalid UUIDs in a replay_id query should now return HTTP 400. 
_get_replay_id_mappings validates the user-provided replay_ids, and raises ValueError which is caught by the endpoint.

TODO: find an example of a 32 char hex that is NOT a UUID?

### CODE REVIEW

**Review by JoshFerge — COMMENTED** [review]
let's add a regression test and make sure that it raises the right error

**Comment by aliu-at-sentry** [comment]
> let's add a regression test and make sure that it raises the right error

On it tomorrow!

File: src/sentry/replays/usecases/replay_counts.py [hunk 1]
```diff
@@ -33,11 +34,19 @@ def get_replay_counts(snuba_params: SnubaParams, query, return_ids, data_source)
     if not replay_ids_mapping:
         return {}
 
+    replay_ids = list(replay_ids_mapping.keys())
+    """ Validate UUIDs. In clickhouse they are stored in their 128-bit int repr.
+     To validate, try converting to the text repr """
+    for int_id in replay_ids:
```
Comment (JoshFerge) [inline src/sentry/replays/usecases/replay_counts.py:40]: i would move this logic into its own function `_validate_uuids` or somesuch.

**Comment by aliu-at-sentry** [comment]
Failing checks on
test_organization_metrics_details.py
test_organization_metrics_tag_details.py
test_organization_metrics_tags.py
which all pass locally. Any insights on this?

**Comment by JoshFerge** [comment]
@aliu-at-sentry its likely nothing to do with your branch. usually re-merging with master can fix. otherwise CI is broken atm

**Review by JoshFerge — APPROVED** [review]
awesome, great stuff

**Pull request merged**